Project-Descriptions-and-Plans/CV32E4_TBD/Project_Concept_for_CV32E4_TBD_Nov-27_2023.md

```diff
@@ -0,0 +1,250 @@
+* Multiply-accumulate
+* General ALU extensions
+* Bit manipulation
+* SIMD (16 and 8-bit data)
+note : this dual opcode scheme allows backard compatibility and open the door to smooth transition
+of existing programs from XPULP to RVB/RVP, if needed/relevant.
+
+Add support for RISC-V official bit manipulation instructions:
+* RVB zba (address manipulation) will be supported
+* RVB zbb (basic bit manipulations) will be supported
+* RVB zbs (single bit instructions) will be supported
+* RVB zbc (carry-less multiplications) : may be supported, to be confirmed if relevant for DSP applications
+* RVK zbkb (Bitmanip instructions for Cryptography) : pack, packh, packw, brev8, zip, unzip additionnal instructions may be supported, if relevant for DSP applications.
+* RVK zbkx (Crossbar permutation instructions) : xpem4 and xperm8 additionnal instructions may be supported, if relevant for DSP applications.
+* RVK zkn, zks, zkr and zkt are specific to cryptographic computations and will not be supported
+
+**Optionnal design changes**
+
+Introduce XPULPv3 custom extentions (preliminary, feature list may evolve):
+* ALU instructions that have a strict RVB/RVP equivalent opcode will be deprecated, new XPULPv3 opcodes will be RVB/RVP ones instead
+* Deprecate single operand *abs* instruction in favor of dual operand *pdif* (RVP superset of abs useful for manhattan distance computation)
+* Deprecate single operand *cv.cplxconj* instruction in favor of dual operand *pas* (RVP parallel add-sub)
+* Deprecate *cv.cplxmul.r* and *cv.cplxmul.i* in favor of a single instruction *cv.cplxmul* that computes both parts of complex multiplication
+* Introduce hermitian form of *cv.cplxmul* (rs1 x conj(rs2)) ? (widely used in DSP computation for telecom applications)
+* *list to be completed / updated*
+
+Add CV-X-IF interface:
+* End user product may require additionnal appliance-specific custom instructions, these instructions should be implemented in a co-processor.
+
+Floating Point Unit:
+* keep current CV-FPU integration or move it to CV-X-IF?
+
+Add CLIC support:
+* Core-local interrupt controller fits better with embedded real-time requirements than PLIC or any other (non RISC-V compliant)
+  external interrupt controller that requires a 2-step interrupt handling code.
+
+Design Portability:
+* Replace global clock gate by equivalent global enable signal
+  * this will make the design more "FPGA-friendly", or compliant with ASIC design rules that forbid any logic addition on clock tree
+  * clock gate cell model and/or simulator behaviour may introduce delta cycles on clock tree, leading to RTL/GL simulation mismatch
+* EDA tools parser bug work-arounds: current design coding style is standard-compliant, but EDA tools may
+  have bugs that prevent proper reading. Design should be parsed with several EDA tools and offending RTL code should be fixed.
+
+### Potential future design enhancements
```

Comment:
We should probably rename this heading in the template as it isn't quite clear. Anyways, are these features thinks you may decide to include within the overall scope of the Project Launch, or are they things that might be added in a future release of this project?

Reply:
These are things that were discussed with several partners. I don't want them being part of the first release of the core for various reasons (reuse of verification code, backward SW compatibility, CV-X-IF specification maturity...) but that should be considered for the next release.